python/lsst/ts/ess/common/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -0,0 +1,377 @@
+# This file is part of ts_ess_common.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["SiglentSSA3000xSpectrumAnalyzerDataClient"]
+
+import asyncio
+import enum
+import logging
+import types
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+import yaml
+from astropy import units
+from lsst.ts import tcpip, utils
+
+from .base_read_loop_data_client import BaseReadLoopDataClient
+
+if TYPE_CHECKING:
+    from lsst.ts import salobj
+
+# The standard TCP/IP line terminator (bytes).
+TERMINATOR = b"\n"
+
+# The expected number of data points as returned by the spectrum analyzer given
+# the fixed start and stop frequency.
+EXPECTED_NUMBER_OF_DATA_POINTS = 751
+
+QUERY_TRACE_DATA_CMD = ":trace:data? 1"
+SET_FREQ_START_CMD = ":frequency:start 0.0 GHz"
+SET_FREQ_STOP_CMD = ":frequency:stop 3.0 GHz"
+
+
+class FreqUnit(enum.Enum):
+    GHz = enum.auto()
+    MHz = enum.auto()
+    kHz = enum.auto()
+    Hz = enum.auto()
+
+
+class SiglentSSA3000xSpectrumAnalyzerDataClient(BaseReadLoopDataClient):
+    """Get data from a Siglent SSA3000X Spectrum analyzer.
+
+    Parameters
+    ----------
+    config : types.SimpleNamespace
+        The configuration, after validation by the schema returned
+        by `get_config_schema` and conversion to a types.SimpleNamespace.
+    topics : `salobj.Controller` or `types.SimpleNamespace`
+        The telemetry topics this data client can write,
+        as a struct with attributes such as ``tel_spectrumAnalyzer``.
+    log : `logging.Logger`
+        Logger.
+    simulation_mode : `int`, optional
+        Simulation mode; 0 for normal operation.
+    """
+
+    ###########################################################################
+    # The following constants need to be hard coded because of limitations in #
+    # the way DDS handles arrays. This ensures that all arrays always have    #
+    # the same, fixed, length.                                                #
+    ###########################################################################
```

Comment:
As we are not using DDS anymore maybe this should be updated? or add a TODO referencing a ticket to adjust it?

Reply:
Thanks for mentioning this. The comment actually is void since the constants it refers to were removed a log time ago. Therefore, I have simply removed the comment.